cmd/monaco/delete/delete.go

```diff
@@ -36,10 +36,10 @@ import (
 // Returns:
 //   - error: If an error occurs during the deletion process, an error is returned, describing the issue.
 //     If no errors occur, nil is returned.
-func Delete(environments manifest.Environments, entriesToDelete delete.DeleteEntries) error {
+func Delete(ctx context.Context, environments manifest.Environments, entriesToDelete delete.DeleteEntries) error {
 	var envsWithDeleteErrs []string
 	for _, env := range environments {
-		ctx := context.WithValue(context.TODO(), log.CtxKeyEnv{}, log.CtxValEnv{Name: env.Name, Group: env.Group})
+		ctx := context.WithValue(ctx, log.CtxKeyEnv{}, log.CtxValEnv{Name: env.Name, Group: env.Group})
```

Comment:
I'm not a big fan of calling both `ctx`

Reply:
It has a point. Here, with showing, nobody will use the wrong context as `ctx` is the common name for it. 
Let's do refactor of extracting the logic in for loop in a function in a separate PR